CONTRIBUTING.md

```diff
@@ -1,49 +1,179 @@
 
 ### ADR Proposals
 
 Architecture Decision Records (ADRs) may be proposed by any contributors or maintainers of ICS. ADRs are intended to be iterative, and may be merged into `main` while still in a `Proposed` status
 
-## PR Targeting
+## Development Procedure
+
+* The latest state of development is on `main`.
+* `main` must never fail `make lint test`.
+* No `--force` onto `main` (except when reverting a broken commit, which should seldom happen).
+* Create a branch to start work:
+    * Fork the repo (core developers must create a branch directly in the ICS repo),
+    branch from the HEAD of `main`, make some commits, and submit a PR to `main`.
+    * For core developers working within the `interchain-security` repo, follow branch name conventions to ensure clear
+    ownership of branches: `{moniker}/{issue#}-branch-name`.
+    * See [Branching Model](#branching-model-and-release) for more details.
+* Be sure to run `make format` before every commit. The easiest way
+  to do this is have your editor run it for you upon saving a file (most of the editors
+  will do it anyway using a pre-configured setup of the programming language mode).
 
-ICS adheres to the [trunk based development branching model](https://trunkbaseddevelopment.com/).
+Code is merged into main through pull request procedure.
 
-All feature additions and all bug fixes must be targeted against `main`. Exception is for bug fixes which may only be relevant to a previously released version. In that case, the bug fix PR must target against the release branch.
+### Testing
 
-If needed, we will backport a commit from `main` to a release branch with appropriate consideration of versioning.
+Appropriate tests should be written with a new feature, and all existing tests should pass. See [docs/testing.md](./docs/old/testing.md) for more information.
 
-## Development Procedure
+### Pull Requests
+
+Before submitting a pull request:
+
+* merge the latest main `git merge origin/main`,
+* run `make lint test` to ensure that all checks and tests pass.
+
+Then:
+
+1. If you have something to show, **start with a `Draft` PR**. It's good to have early validation of your work and we highly recommend this practice. A Draft PR also indicates to the community that the work is in progress.
+   Draft PRs also help the core team provide early feedback and ensure the work is in the right direction.
+2. When the code is complete, change your PR from `Draft` to `Ready for Review`.
+3. Go through the actions for each checkbox present in the PR template description. The PR actions are automatically provided for each new PR.
+4. Be sure to include a relevant changelog entry in the `Unreleased` section of `CHANGELOG.md` (see file for log format). The entry should be on top of all others changes in the section.
```

Comment:
Once https://github.com/cosmos/interchain-security/issues/801 is completed, we should start using [unclog](https://github.com/informalsystems/unclog).